compiler/src/iree/compiler/InputConversion/StableHLO/test/stablehlo_to_linalg_ext.mlir

```diff
@@ -0,0 +1,546 @@
+// CHECK-DAG:    %[[COEF_REAL:.+]] = arith.constant dense<{{.+}}> : tensor<2xf32>
+// CHECK-DAG:    %[[COEF_IMAG:.+]] = arith.constant dense<{{.+}}> : tensor<2xf32>
+// CHECK:        %[[R2:.+]]:2 = iree_linalg_ext.fft
+// CHECK-SAME:     ins(%[[C2]], %[[COEF_REAL]], %[[COEF_IMAG]]
+// CHECK-SAME:     outs(%[[R1]]#0, %[[R1]]#1
+// CHECK-DAG:    %[[C3:.+]] = arith.constant 3 : index
+// CHECK-DAG:    %[[COEF_REAL:.+]] = arith.constant dense<{{.+}}> : tensor<4xf32>
+// CHECK-DAG:    %[[COEF_IMAG:.+]] = arith.constant dense<{{.+}}> : tensor<4xf32>
+// CHECK:        %[[R3:.+]]:2 = iree_linalg_ext.fft
+// CHECK-SAME:     ins(%[[C3]], %[[COEF_REAL]], %[[COEF_IMAG]]
+// CHECK-SAME:     outs(%[[R2]]#0, %[[R2]]#1
+// CHECK:        %[[RES_REAL:.+]] = tensor.extract_slice %[[R3]]#0[0, 0] [4, 5] [1, 1] : tensor<4x8xf32> to tensor<4x5xf32>
+// CHECK:        %[[RES_IMAG:.+]] = tensor.extract_slice %[[R3]]#1[0, 0] [4, 5] [1, 1] : tensor<4x8xf32> to tensor<4x5xf32>
+// CHECK:        %{{.+}} = stablehlo.complex %[[RES_REAL]], %[[RES_IMAG]]
+
+// -----
+
+// CHECK-LABEL: func.func @reverse_dim1
+// CHECK-SAME:   %[[IN:[a-zA-Z0-9]+]]
+func.func @reverse_dim1(%arg0: tensor<3x5xi32>) -> tensor<3x5xi32> {
+  %0 = "stablehlo.reverse"(%arg0) {
+    dimensions = dense<1> : tensor<1xi64>
+  } : (tensor<3x5xi32>) -> tensor<3x5xi32>
+  return %0 : tensor<3x5xi32>
+}
+// CHECK:        %[[INIT:.+]] = tensor.empty() : tensor<3x5xi32>
+// CHECK:        %[[REV:.+]] = iree_linalg_ext.reverse
+// CHECK-SAME:     dimensions(dense<1> : tensor<1xi64>)
+// CHECK-SAME:     ins(%[[IN]] : tensor<3x5xi32>)
+// CHECK-SAME:     outs(%[[INIT]] : tensor<3x5xi32>) : tensor<3x5xi32>
+// CHECK:        return %[[REV]]
+
+// -----
+
+// CHECK-LABEL: func.func @reverse_multi_dim
+// CHECK-SAME:   %[[IN:[a-zA-Z0-9]+]]
+func.func @reverse_multi_dim(%arg0: tensor<?x?xi32>) -> tensor<?x?xi32> {
+  %0 = "stablehlo.reverse"(%arg0) {
+    dimensions = dense<[0, 1]> : tensor<2xi64>
+  } : (tensor<?x?xi32>) -> tensor<?x?xi32>
+  return %0 : tensor<?x?xi32>
+}
+// CHECK-DAG:    %[[C0:.+]] = arith.constant 0 : index
+// CHECK-DAG:    %[[C1:.+]] = arith.constant 1 : index
+// CHECK-DAG:    %[[D0:.+]] = tensor.dim %[[IN]], %[[C0]]
+// CHECK-DAG:    %[[D1:.+]] = tensor.dim %[[IN]], %[[C1]]
+// CHECK:        %[[INIT:.+]] = tensor.empty(%[[D0]], %[[D1]]) : tensor<?x?xi32>
+// CHECK:        %[[REV:.+]] = iree_linalg_ext.reverse
+// CHECK-SAME:     dimensions(dense<[0, 1]> : tensor<2xi64>)
+// CHECK-SAME:     ins(%[[IN]] : tensor<?x?xi32>)
+// CHECK-SAME:     outs(%[[INIT]] : tensor<?x?xi32>) : tensor<?x?xi32>
+// CHECK:        return %[[REV]]
+
+// -----
+
+// CHECK:       func.func @chlo_top_k_int
+// CHECK-SAME:   %[[ARG0:[a-zA-Z0-9]+]]
+func.func @chlo_top_k_int(%arg : tensor<16x16xi32>) -> (tensor<16x8xi32>, tensor<16x8xi32>) {
+  %1:2 = chlo.top_k(%arg, k=8) : tensor<16x16xi32> -> (tensor<16x8xi32>, tensor<16x8xi32>)
+  return %1#0, %1#1 : tensor<16x8xi32>, tensor<16x8xi32>
+}
```

Comment:
Ah.. `chlo` down here... should we find something in stablehlo instead? Or find another directory for tests?

Reply:
I want to keep them together for now before it becomes clear whether to split or not: https://github.com/openxla/iree/pull/13331#discussion_r1180929684